Makefile

```diff
@@ -33,36 +33,71 @@ nvim: | ${NVIM_PATH}
 	git -C ${NVIM_MASTER_PATH} fetch origin master --depth 1
 	git -C ${NVIM_MASTER_PATH} checkout FETCH_HEAD
 
-LUASNIP_DETECTED_OS?=$(shell uname)
+LUASNIP_DETECTED_OS?=$(shell uname && false)
 ifeq ($(LUASNIP_DETECTED_OS),Darwin)
 	# flags for dynamic linking on macos, from luarocks
 	# (https://github.com/luarocks/luarocks/blob/9a3c5a879849f4f411a96cf1bdc0c4c7e26ade42/src/luarocks/core/cfg.lua#LL468C37-L468C80)
 	# remove -bundle, should be equivalent to the -shared hardcoded by jsregexp.
 	LUA_LDLIBS=-undefined dynamic_lookup -all_load
 endif
 
-JSREGEXP_PATH=deps/jsregexp
-JSREGEXP005_PATH=deps/jsregexp005
+# Windows may have $OS env been set natively
+OS_ENV?=$(OS)
+ifeq ($(OS_ENV),Windows_NT)
+	LUASNIP_DETECTED_OS:=Windows
+endif
+# Example output on Windows: MINGW64_NT-10.0-19045 DESKTOP-ABCDE 3.4.10-1234567.x86_64 2024-02-14 20:17 UTC x86_64 Msys
+UNAME_ALL:=$(shell uname -a && false)
+ifneq (,$(findstring MINGW,$(UNAME_ALL)))
+	LUASNIP_DETECTED_OS:=Windows
+endif
+ifneq (,$(findstring Msys,$(UNAME_ALL)))
+	LUASNIP_DETECTED_OS:=Windows
+endif
+
+# On Windows, you may need to set:
+# 	SHELL=C:/path/to/Git/usr/bin/sh.exe
+# 	.SHELLFLAGS=-c
+# 	CC=gcc
+#	NEOVIM_BIN_PATH=C:/path/to/Neovim/bin # contains lua51.dll, or use your own LUA_LDLIBS
+ifeq ($(LUASNIP_DETECTED_OS),Windows)
+	# If neovim is installed by scoop, only scoop/shims is exposed. We need to find original nvim/bin that contains lua51.dll
+	# If neovim is installed by winget or other methods, nvim/bin is already included in PATH.
+	NEOVIM_BIN_PATH?=$(shell \
+		if (scoop prefix neovim | grep -v 'Could not find app path for') >/dev/null 2>&1; then \
+			echo "$$(scoop prefix neovim)/bin" | sed 's/\\\\/\\//g' | sed 's/\\(.*\\) \\(.*\\)/"\\1 \\2"/'; \
+		elif which nvim > /dev/null 2>&1; then \
+			dirname "$$(which nvim)" | sed 's/^\\/\\(.\\)\\//\\U\\1:\\//' | sed 's/\\(.*\\) \\(.*\\)/"\\1 \\2"/'; \
+		fi)
+
+	LUA_LDLIBS?=$(if $(strip $(NEOVIM_BIN_PATH)),-L$(NEOVIM_BIN_PATH) -llua51,)
+endif
+
+# some workaround to empty $(shell command) output on Windows
+# @link: https://stackoverflow.com/questions/74481637/shell-command-in-makefile-returns-empty-on-windows-sometimes
+PROJECT_ROOT:=$(shell pwd && false)
```

Comment:
Could you educate me on this? I read the linked thread, but  don't really see a solution to the problem :sweat_smile: